File: 2015/py/day09.py
import sys
from collections import defaultdict
from itertools import permutations


def main():
    part1 = sys.maxsize
    part2 = 0

    graph = defaultdict(dict)

    for line in sys.stdin:
        start, end, distance = line.split(" ")[::2]
        graph[start][end] = graph[end][start] = int(distance)

    for route in permutations(graph.keys()):
        distance = sum(graph[a][b] for a, b in zip(route, route[1:]))

        part1 = min(part1, distance)
        part2 = max(part2, distance)

    print(part1)
    print(part2)


if __name__ == "__main__":
    main()

File: 2015/rs/src/bin/day09.rs
use std::collections::HashMap;
use std::io;

fn read_graph() -> HashMap<(String, String), u32> {
    let mut graph = HashMap::new();

    let lines = io::stdin()
        .lines()
        .map(|result| result.expect("Failed to read line from stdin"));

    for _line in lines {
        graph.insert(("foo".to_string(), "bar".to_string()), 10);
    }

    graph
}

fn main() {
    let mut part1 = 0;
    let mut part2 = 0;

    let graph = read_graph();

    println!("{part1}");
    println!("{part2}");
}
